src/pages/index.js

```diff
@@ -4,78 +4,224 @@ import FormValidator from "../components/FormValidator.js";
+      } else {
+        console.error('cardSection or addItem method is undefined.');
+      }
+      
+    })
+    .catch((error) => {
+      console.error('Error adding new card:', error);
+      // Reset loading state in case of an error
+    });
+}
+  
+// Update avatar function
+function handleAvatarUpdate(formData) {
+  updateAvatarPopup.setLoading(true);
+
+  // Get the link value from the input field
+  // const linkInput = document.getElementById('new-avatar-link-input');
+  // const link = linkInput.value;
+  const avatarUrl = formData.link;
+
+  // Ensure api.updateAvatar exists
+  if (typeof api.updateAvatar !== 'function') {
+    console.error('updateAvatar function is not defined in the API');
+    // updateAvatarPopup.setLoading(false); // Reset loading state in case of error
+    return;
+  }
+
+  // Call the API to update the avatar
+  api.updateAvatar(avatarUrl)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      updateAvatarPopup.setLoading(false);
+      updateAvatarPopup.close(); // Ensure updateAvatarPopup is the correct reference
+    })
+    .catch((error) => {
+      console.error(error);
+      updateAvatarPopup.setLoading(false); // Reset loading state in case of error
+    });
+}
+
+// Delete card function
+function handleDeleteCard(card) {
+  confirmModal.open();
+
+  confirmModal.setSubmitAction(() => {
+    api.deleteCard(card.id)
+    .then(() => {
+      card.deleteCard();
+      confirmModal.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+  })
+}
+
+// Toggle like/dislike on card function
+function handleLikeCard(card) {
+  return api.toggleLikeCard(card.getId(), !card.getIsLiked())
+    .then((updatedCardData) => {
+      card.updateLikeValue(updatedCardData.isLiked);
+      return updatedCardData; // Assuming the API returns the updated card data
+    })
+    .catch((error) => {
+      console.log(error);
+      throw error; // Re-throw the error to be handled by the caller
+      
+    });
+}
+
+// Profile edit function
+function handleProfileEditSubmit(values) {
+  // Get the new profile data from the form inputs
+  const name = profileTitleInput.value;
+  const about = profileDescriptionInput.value;
+
+ editProfilePopup.setLoading(true);
+
+   // Call the API to update user info
+   api.updateUserInfo(name, about)
+   .then((userData) => {
+     userInfo.setUserInfo(userData); // Update the UI with the new profile data
+     editProfilePopup.setLoading(false); 
+     editProfilePopup.close();
+   })
+   .catch((error) => {
+     console.log(error);
+    //  editprofilepopup.setLoading(false); // Set loading state to false even if there's an error
+   });
+
+}
 
 // Wrappers
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const profileEditForm = profileEditModal.querySelector("#profile-edit-form");
 const addCardModal = document.querySelector("#add-card-modal");
 const addCardForm = addCardModal.querySelector("#add-card-form");
```

Comment:
**Could be improved**

We can get any form using their **unique** attribute `name` or `id` from `document.forms`

```js

const profileForm = document.forms["profile-form"];
const cardForm = document.forms["card-form"];

```

You need to add `name` or `id` first in `index.html` to achieve that.

More info here https://developer.mozilla.org/en-US/docs/Web/API/Document/forms